Doc/reference/expressions.rst

```diff
@@ -1054,10 +1054,20 @@ used in the same call, so in practice this confusion does not often arise.
 
 If the syntax ``**expression`` appears in the function call, ``expression`` must
 evaluate to a :term:`mapping`, the contents of which are treated as
-additional keyword arguments.  If a keyword is already present
-(as an explicit keyword argument, or from another unpacking),
+additional keyword arguments. If a parameter matching a key has already been
+given a value (by an explicit keyword argument, or from another unpacking),
 a :exc:`TypeError` exception is raised.
 
+When ``**expression`` is used, each key in this mapping must be
+an instance of :class:`str` or a :term:`hashable` sub-class of it.
```

Comment:
We don't need to mention subclasses -- the phrase "an instance of" implies subclassing. I also don't think we need to mention hashable -- that is only needed when the callee uses `**kwds` in its formal parameter list.
```suggestion
an instance of :class:`str`.
```

Reply:
I agree that "instance of" includes instances of sub-classes.

Am I right that the formal parameters of a callable must be identifiers? There's no other way to create them in Python?

In that case, being "hashable" (meeting the contract of `__eq__` and `__hash__`) seems relevant. Keys that are *not* valid identifiers cannot match a formal parameter and *must* end up in the callee's `**kwds` (if it has one) or lead to a `TypeError` (not the interesting case).

Reply:
> I agree that "instance of" includes instances of sub-classes.

Good.

> Am I right that the formal parameters of a callable must be identifiers? There's no other way to create them in Python?

That sounds like a trick question. :-)

> In that case, being "hashable" (meeting the contract of `__eq__` and `__hash__`) seems relevant. Keys that are _not_ valid identifiers cannot match a formal parameter and _must_ end up in the callee's `**kwds` (if it has one) or lead to a `TypeError` (not the interesting case).

But we’re not trying to give an exhaustive specification here. A subclass of str that fails to implement that contract is simply faulty. There’s nothing interesting to specify or check here. (Heaven forbid someone might interpret your wording as an indication that they should explicitly check for hashability.)

Reply:
If "instance of `str`" says it all, I'll go with the suggestion.

While we're both here, should I try to do the same for `getattr()` and `__dict__` within this  PR or separately (or not at all)?